### ISSUE TITLE
Add swagger api document

### ISSUE THREAD

**Issue by narrowizard** [apache/incubator-devlake#292]
## Description
We should add document for restful api.

## Describe the solution you'd like
Recomended to use [swagger](https://swagger.io/) with a lot great features and extensions, for example: 
- generator documents from code with [swaggo](https://github.com/swaggo/swag)
- auto resolve apis with [gin-swagger](https://github.com/swaggo/gin-swagger)
- an easy-use web UI
![image](https://user-images.githubusercontent.com/11383928/130886860-31a655d2-d6c9-49a3-b51d-98d9b53c3898.png)
- try out apis from web UI
![image](https://user-images.githubusercontent.com/11383928/130887112-fb134af1-40dd-479b-81e5-e3841e76df22.png)

**Comment by xgdyp** [comment]
I'll pick it！

**Comment by xgdyp** [comment]
I noticed the PR before was closed.  Should I use go-swagger or just write down the docs?

**Comment by klesh** [comment]
Just a note: as we discussed on IM the other day, it was closed due to some implementation  issues. We need this sooner or later.
We should try to accomplish the following targets regarding implementation:
1. Can swagger server run along side api server? AKA, under different `path` but same `port`?
2. Can document be generated automatically from the source code, not a separated `json` file?
3. Can we call api directly from swagger doc?
